### ISSUE TITLE
"Use of moved value" false positive

### ISSUE THREAD

**Issue by everesio** [intellij-rust/intellij-rust#5013]
## Environment

* **IntelliJ Rust plugin version: 0.2.116.2163-193**
* **Rust toolchain version: v1.42.0**
* **IDE name and version: IntelliJ IDEA 2019.3.3**
* **Operating system: Ubuntu 19.10**

## Problem description
The main() function below compiles but is incorrectly flagged as a "Use of a moved value" on the "Destructure the point using a `let` binding". 
The snippet comes from https://doc.rust-lang.org/rust-by-example/custom_types/structs.html

## Steps to reproduce

```rust
// A struct with two fields
struct Point {
    x: f32,
    y: f32,
}

fn main() {

    // Instantiate a `Point`
    let point: Point = Point { x: 10.3, y: 0.4 };

    // Access the fields of the point
    println!("point coordinates: ({}, {})", point.x, point.y);

    // Make a new point by using struct update syntax to use the fields of our
    // other one
    let bottom_right = Point { x: 5.2, ..point };

    // Destructure the point using a `let` binding
    let Point { x: top_edge, y: left_edge } = point;
}
```

**Comment by dvdplm** [comment]
Seeing this too on `0.2.117.3001-201-nightly` and CLion 2020.1 EAP, for instance [here](https://github.com/OpenEthereum/open-ethereum/blob/77643c13e80ca09d9a6b10631034f5a1568ba6d3/ethcore/state-db/src/lib.rs#L174) but almost every file in that repo has the same problem.
